src/provider/controller/NotificationCenter.ts

```diff
@@ -1,145 +1,220 @@
-import {TrayIconClicked} from 'openfin/_v2/api/events/application';
-
-import {WindowInfo} from './WindowInfo';
-declare const window: Window & {WindowManager: NotificationCenter};
-
+import {WindowOption} from 'openfin/_v2/api/window/windowOption';
+import {Store} from 'redux';
+import {MonitorEvent} from 'openfin/_v2/api/events/system';
+
+import {WebWindow, createWebWindow} from '../model/WebWindow';
+import {renderApp} from '../view/containers/NotificationCenterApp';
+import {TrayIcon} from '../common/TrayIcon';
+import {watchForChange} from '../store/utils/watch';
+import {getNotificationCenterVisibility} from '../store/ui/selectors';
+import {toggleCenterWindowVisibility, changeBannerDirection} from '../store/ui/actions';
+import {RootState} from '../store/typings';
+import {ContextMenu, MenuItem} from '../common/ContextMenu';
+
+const windowOptions: WindowOption = {
+    name: 'Notification-Center',
+    url: 'ui/index.html',
+    autoShow: true,
+    defaultHeight: 400,
+    defaultWidth: 500,
+    resizable: false,
+    saveWindowState: false,
+    defaultTop: 0,
+    contextMenu: !(process.env.NODE_ENV === 'production'),
+    frame: false,
+    alwaysOnTop: true,
+    icon: 'ui/favicon.ico',
+    showTaskbarIcon: false,
+    opacity: 0
+};
+
+interface Options {
+    /** Blur event causes window to hide */
+    hideOnBlur?: boolean;
+}
 
 export class NotificationCenter {
-    private windowInfo: WindowInfo = new WindowInfo();
-    private static singleton: NotificationCenter;
-
-    constructor() {
-        if (NotificationCenter.singleton) {
-            return NotificationCenter.singleton;
-        }
-        this.setEventListeners();
-        NotificationCenter.singleton = this;
-        this.setupTrayIcon();
-    }
-
-    /**
-     * @function setEventListeners Initalizes event listeners for the window
-     * @returns void
-     */
-    public setEventListeners(): void {
-        // When window is ready
-        window.addEventListener('DOMContentLoaded', this.onWindowLoad.bind(this));
-
-        // On window close requested
-        this.windowInfo.getWindow().addEventListener('close-requested', () => this.hideWindow());
-
-        // On monitor dimension change
-        fin.desktop.System.addEventListener('monitor-info-changed', this.onMonitorInfoChanged.bind(this));
+    private _webWindow: Promise<WebWindow>;
+    private _trayIcon!: TrayIcon;
+    private _options: Options;
+    private _store: Store;
+
+    public constructor(store: Store<RootState>, options: Options) {
```

Comment:
Having dependency injection will be useful to for injecting the `ToastManager` & `NotificationsCenter` into the provider middleware instead of using singletons.